src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,261 @@
+
+public class Duke {
+
+    private static final String LOGO = " ____        _        \n"
+                                    + "|  _ \\ _   _| | _____ \n"
+                                    + "| | | | | | | |/ / _ \\\n"
+                                    + "| |_| | |_| |   <  __/\n"
+                                    + "|____/ \\__,_|_|\\_\\___|\n";
+    private static final String LINE_SEPARATOR = "------------------------------------";
+    private static final String HELLO_MESSAGE = "Hello! I'm duke.Duke\nWhat can I do for you?";
+    private static final String BYE_MESSAGE = "Bye. Hope to see you again soon!";
+    private static final String ERROR_MESSAGE = "You need to specify the task type!";
+
+    private static final String FILE_PATH = "data/duke.txt";
+
+    private static final String GAP = " / ";
+
+    private static ArrayList<Task> tasks = new ArrayList<>();
+
+    public static void main(String[] args) {
+        loadTasks();
+        printHelloMessage();
+        handleCommand();
+        printByeMessage();
+        saveTasks();
+    }
+
+    private static void loadTasks() {
+        try {
+            addTasksIntoList();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    private static File loadFile() throws IOException {
+        File file = new File(FILE_PATH);
+        if (!file.exists()) {
+            File dir = new File("data");
+            dir.mkdir();
+            File newFile = new File("data/duke.txt");
+            newFile.createNewFile();
+            return newFile;
+        }
+        return file;
+    }
+
+    private static void addTasksIntoList() throws IOException {
+        File f = loadFile();
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String entry = s.nextLine();
+            String[] entryComponents = entry.split(GAP);
+            String description = entryComponents[0] + " " + entryComponents[2];
+            switch (entryComponents[0]) {
+            case "T":
+                tasks.add(new Todo(description));
+                break;
+            case "D":
+                String by = "by " + entryComponents[3];
+                tasks.add(new Deadline(description, by));
+                break;
+            case "E":
+                String at = "at " + entryComponents[3];
+                tasks.add(new Event(description, at));
+                break;
+            default:
+                break;
+            }
+            if (entryComponents[1].equals("X")) {
+                tasks.get(tasks.size() - 1).completeTask();;
+            }
+        }
+    }
+
+    private static void saveTasks() {
+        try {
+            writeTasksToFile();
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    private static void writeTasksToFile() throws IOException {
+        FileWriter fw = new FileWriter(FILE_PATH);
```

Comment:
Please consider using names like "fileWriter" for the FileWriter object instead of "fw".